internal/cloud/gcp.go

```diff
@@ -116,11 +116,12 @@ func (gcp *GCP) MountBucket(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodS
 	return fmt.Errorf("container not found: %s", req.Container)
 }
 
-func (gcp *GCP) AssociateServiceAccount(sa *corev1.ServiceAccount) {
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
-	}
-	sa.Annotations["iam.gke.io/gcp-service-account"] = fmt.Sprintf("substratus-%s@%s.iam.gserviceaccount.com", sa.Name, gcp.ProjectID)
+func (gcp *GCP) GetPrincipal(sa *corev1.ServiceAccount) string {
+	return fmt.Sprintf("substratus@%s.iam.gserviceaccount.com", gcp.ProjectID)
```

Comment:
This will result in the principal set in `PRINCIPAL` env variable never getting used as this method will overshadow the one from `Common`. Prefer to either leave this method off of `GCP` and lean on the one from `Common`. If you want to default to this email determined from `gcp.ProjectID` it should be done in `AutoConfigure()`. See how `RegistryURL` is set for reference.

Reply:
Fixed by using AutoConfigure. Thanks for catching this!